app/controllers/application_controller/ci_processing.rb

```diff
@@ -978,6 +970,30 @@ def process_elements(elements, klass, task, display_name = nil, order_field = ni
 
   private ############################
 
+  def process_element_destroy(element, klass, name)
+    return unless element.respond_to?(:destroy)
+
+    audit = {:event        => "#{klass.name.downcase}_record_delete",
+             :message      => "[#{name}] Record deleted",
+             :target_id    => element.id,
+             :target_class => klass.base_class.name,
+             :userid       => session[:userid]}
+
+    model_name  = ui_lookup(:model => klass.name) # Lookup friendly model name in dictionary
+    record_name = get_record_display_name(element)
+
+    element.destroy
```

Comment:
Should we wrap this in a begin/rescue (as originally done in the caller method) in case the destroy raises an error or is it guaranteed to go down the code path below in all situations?  /cc @Fryguy